packages/celer-web-app/src/app/AppDocumentProvider.tsx

```diff
@@ -17,71 +18,81 @@ import {
 import { ServiceCreator } from "./services";
 
 export type AppDocumentProviderProps = {
-    serviceCreator: ServiceCreator,
-    //still need this for local document...
-    shouldSetBundle: boolean
+	serviceCreator: ServiceCreator,
+	//still need this for local document...
+	shouldSetBundle: boolean
 }
 
 const compiler = new Compiler();
 const routeEngine = new RouteEngine();
 const mapEngine = new MapEngine();
 
-export const AppDocumentProvider: React.FC<AppDocumentProviderProps> = ({serviceCreator, shouldSetBundle, children}) => {
+export const AppDocumentProvider: React.FC<AppDocumentProviderProps> = ({ serviceCreator, shouldSetBundle, children }) => {
 	const warnNegativeVar = useExpWarnNegativeVar();
 	const enableInferCoord = useExpInferCoord();
 	const enableBetterBundler = useExpBetterBundler();
 
 	const { splitSetting } = useAppSetting();
 
-	useEffect(()=>{
+	useEffect(() => {
 		routeEngine.warnNegativeNumberEnable = warnNegativeVar;
 		routeEngine.inferCoord = enableInferCoord;
 	}, [warnNegativeVar, enableInferCoord]);
-    
+
 	const params = useParams();
-	const [status, setStatus] = useState<string|null>(null);
-	const [error, setError] = useState<string|null>(null);
-	const [routeSourceBundle, setRouteSourceBundle] = useState<SourceObject|null>(null);
-	const [routeSourceBundleString, setRouteSourceBundleString] = useState<string|null>(null);
+	const [status, setStatus] = useState<string | null>(null);
+	const [error, setError] = useState<string | null>(null);
+	const [routeSourceBundle, setRouteSourceBundle] = useState<SourceObject | null>(null);
+	const [routeSourceBundleString, setRouteSourceBundleString] = useState<string | null>(null);
 
-	useEffect(()=>{
+	useEffect(() => {
 		const service = serviceCreator(params);
-		service.start((doc, error, status)=>{
-			if(doc){
-				if(doc._globalError){
+		service.start((doc, error, status) => {
+			if (doc) {
+				if (doc._globalError) {
 					setError(doc._globalError);
 					setStatus(null);
 					setRouteSourceBundle(null);
-				}else{
+				} else {
 					setError(null);
 					setStatus(null);
 					setRouteSourceBundle(doc);
+					// Store current page in localStorge if it is a GitHub route
+					if (("user" in params) && ("repo" in params)) {
+						// Define the current url from the parameters
+						let url = "gh/" + params["user"] + "/" + params["repo"];
```

Comment:
Ok.. not future proof. What if I need to add another service like `#abc/{:user}/{:repo}` in the future?
Another problem is that if someone needs to change the `gh` route, he/she might not know that this place also needs to be updated, which would cause a bug.
I think it would be better if we can define something in the service interface and let it return the current document path